src/accountingservice/main.go

```diff
@@ -76,16 +78,34 @@ func initTracerProvider() (*sdktrace.TracerProvider, error) {
 	return tp, nil
 }
 
+func initDDTracerProvider() *ddotel.TracerProvider {
+	tp := ddotel.NewTracerProvider(
+		ddtracer.WithAgentAddr(fmt.Sprintf("%s:%s", os.Getenv("HOST_IP"), "8126")),
+	)
+	otel.SetTracerProvider(tp)
+	otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}))
+	return tp
+}
+
 func main() {
-	tp, err := initTracerProvider()
-	if err != nil {
-		log.Fatal(err)
-	}
-	defer func() {
-		if err := tp.Shutdown(context.Background()); err != nil {
-			log.Printf("Error shutting down tracer provider: %v", err)
+	if envTag := os.Getenv("OTEL_K8S_NAMESPACE"); envTag == "otel-ingest-staging" {
```

Comment:
Instead of envTag, it would be better if we check `DD_AGENT_HOST` env. If it present use DD tracer.

Reply:
That requires if-else condition in the helm values.yaml, I don't see that's clearer than the current check.

Currently `DD_AGENT_HOST` is set in all namespaces but only `otel-ingest-staging` makes use of it.